fe/topology.py

```diff
@@ -390,6 +555,163 @@ def parameterize_improper_torsion(self, ff_params):
+
+        num_torsions = torsion_params.shape[0]
+
+        lambda_mult_idxs = np.zeros(num_torsions, dtype=np.int32)
+        lambda_offset_idxs = np.ones(num_torsions, dtype=np.int32)
+
+        for torsion_idx, (_, b, c, _) in enumerate(torsion_potential.get_idxs()):
+            if membership[b] != membership[c]:
+                lambda_offset_idxs[torsion_idx] = 0
+
+        torsion_potential.set_lambda_mult_and_offset(lambda_mult_idxs, lambda_offset_idxs)
+
+        return torsion_params, torsion_potential
+
+    def parameterize_nonbonded(self,
+        ff_q_params,
+        ff_lj_params):
+
+        # both mol_a and mol_b are standardized.
+        # we don't actually need derivatives for this stage.
+
+        # zero out/stop grad
+        ff_q_params = np.zeros_like(ff_q_params)
+        ff_lj_params = np.zeros_like(ff_lj_params)
+
+        qlj_params, nb_potential = super().parameterize_nonbonded(ff_q_params, ff_lj_params)
+        qlj_params = jax.ops.index_update(qlj_params, jax.ops.index[:, 0], _STANDARD_CHARGE)
+        mol_c = Chem.CombineMols(self.mol_a, self.mol_b)
+        sig_eps = simple_lj_typer(mol_c)
+        qlj_params = jax.ops.index_update(qlj_params, jax.ops.index[:, 1], sig_eps[:, 0])
+        qlj_params = jax.ops.index_update(qlj_params, jax.ops.index[:, 2], sig_eps[:, 1])
+
+        # ligand is already decharged by now
+        # super-ligand state should have half the epsilons
+        src_qlj_params = jax.ops.index_update(qlj_params, jax.ops.index[:, 2], qlj_params[:, 2]*0.5)
+        dst_qlj_params = qlj_params
+
+        combined_qlj_params = jnp.concatenate([src_qlj_params, dst_qlj_params])
+
+        interpolated_potential = nb_potential.interpolate()
+        combined_lambda_plane_idxs = np.zeros(
+            self.mol_a.GetNumAtoms() + self.mol_b.GetNumAtoms(),
+            dtype=np.int32
+        )
+        combined_lambda_offset_idxs = np.concatenate([
+            np.zeros(self.mol_a.GetNumAtoms(), dtype=np.int32),
+            np.ones(self.mol_b.GetNumAtoms(), dtype=np.int32)
+        ])
+        interpolated_potential.set_lambda_plane_idxs(combined_lambda_plane_idxs)
+        interpolated_potential.set_lambda_offset_idxs(combined_lambda_offset_idxs)
+
+        return combined_qlj_params, interpolated_potential
+
+
+class DualTopologyMinimization(DualTopology):
+
+    def parameterize_nonbonded(self,
+        ff_q_params,
+        ff_lj_params):
+
+        # both mol_a and mol_b are standardized.
+        # we don't actually need derivatives for this stage.
+        qlj_params, nb_potential = super().parameterize_nonbonded(ff_q_params, ff_lj_params)
+
+        combined_lambda_plane_idxs = np.zeros(
+            self.mol_a.GetNumAtoms() + self.mol_b.GetNumAtoms(),
+            dtype=np.int32
+        )
+        combined_lambda_offset_idxs = np.concatenate([
+            np.ones(self.mol_a.GetNumAtoms(), dtype=np.int32),
+            np.ones(self.mol_b.GetNumAtoms(), dtype=np.int32)
+        ])
```

Comment:
```suggestion
        n_A, n_B = self.mol_a.GetNumAtoms(), self.mol_b.GetNumAtoms()
        combined_lambda_plane_idxs = np.zeros(n_A + n_B, dtype=np.int32)
        combined_lambda_offset_idxs = np.ones(n_A + n_B, dtype=np.int32)
```

Reply:
Will edit this in directly with a minor change on capitalization of the N